File: Java/TP3_MET_JAVA/src/Inductif/TailleInductif.java
package Inductif;

public class TailleInductif implements VisiteurMotInductif<Integer> {

    @Override
    public Integer casVide() {
        return 0;
    }

    @Override
    public Integer casCaractereMot(Character c, Integer m) {
        return m + 1;
    }
}

File: TypeScript/TP3_MET_TS/src/Inductif/TailleInductif.ts
import {VisiteurMotInductif} from "./VisiteurMotInductif";

export class TailleInductif implements VisiteurMotInductif<number> {

    public casVide(): number {
        return 0;
    }

    public casCaractereMot(c: string, m: number): number {
        return m + 1;
    }
}
